packages/@aws-cdk/aws-lambda-nodejs/lib/bundling.ts

```diff
@@ -122,21 +134,38 @@ export class Bundling implements cdk.BundlingOptions {
 
   private createBundlingCommand(options: BundlingCommandOptions): string {
     const pathJoin = osPathJoin(options.osPlatform);
+    const npx = options.osPlatform === 'win32' ? 'npx.cmd' : 'npx';
+
+    let compileCommand = '';
+    let relativeJsEntryPath;
+
+    if (this.preCompilation) {
+      relativeJsEntryPath = pathJoin(options.inputDir, PRE_COMPILATION_DIR, this.relativeEntryPath)
+        .replace(/\.(tsx)$/, '.jsx').replace(/\.(ts)$/, '.js');
+
+      if (this.relativeTsconfigPath) {
+        const tsconfigPath = pathJoin(options.inputDir, this.relativeTsconfigPath);
+        const rootDir = extractRootDir(tsconfigPath);
+        compileCommand =`${npx} tsc --project ${tsconfigPath} --outDir ${pathJoin(options.inputDir, PRE_COMPILATION_DIR, rootDir ?? '')}`;
```

Comment:
I wonder if it wouldn't be better to compile without `--outDir` (= next to .ts files). What do you think?

Also, this will compile all ts files right? Even CDK/infrastructure files? Is there a way we can optimize this?

Reply:
This will indeed compile all the files that exists in the project even the cdk/infrastructure files too, but the current implementation doesn't enforce anything related to that, this really depends on the configuration of users tsconfig.

The improvement I can see here is that we should run tsc once only, currently its running each time when there is q request to bundle.

Reply:
I do like the suggestion here of compiling next to .ts files, in that we are simply ignoring users configuration that exists in tsconfig.